### PR TITLE
Create documentation page about events 

### PR DESCRIPTION
**Pull request by callumforrester**
* Add page about events to developer documentation
* Explain why we need to add new events on top of the event model
* Explain the use of correlation IDs

### CODE REVIEW

**Comment by callumforrester** [comment]
Still to do: make API reference and point to that

File: docs/developer/explanations/events.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
+Events Emitted by the Worker
+============================
+
+Blueapi adds new events on top of the `bluesky event model`_.
+
+Reasons
+-------
+
+Since the ``RunEngine`` is traditionally used by a human in front of an IPython terminal, it
+sometimes assumes intuitive behavoir. The worker replaces the human and so must fill in the
```
Comment (DiamondJoseph) [inline docs/developer/explanations/events.rst:10]: behavoir -> behaviour

**Review by DiamondJoseph — CHANGES_REQUESTED** [review]
nit, otherwise LGTM

**Comment by DiamondJoseph** [comment]
Presumably going to need an update for emitting state events on same topic?

**Comment by keithralphs** [comment]
> Presumably going to need an update for emitting state events on same topic?

Do that as part of addressing #176

**Pull request merged**